src/detect-cipservice.c

```diff
@@ -157,7 +157,10 @@ static DetectCipServiceData *DetectCipServiceParse(const char *rulestrc)
             goto error;
         }
 
-        sscanf(token, "%2" SCNu8, &var);
+        if (sscanf(token, "%2" SCNu8, &var) != 1) {
```

Comment:
it looks like the original code has some serious issues, c5cf2967b3bc7fc074052279fe35d29fae283f43 broke things by turning the `input` array into a `uint8_t` array, even though for class and attribute we're supposed to have a `uint16_t` values. So that would need fixing.

In general, on further inspection, it's unclear why we have the `sscanf` at all. We already have `num`, and it's value is validated to be in range, so we can safely cast it to `uint8_t` for service and `uint16_t` for the other two.

cc @catenacyber

Reply:
Complementing this: a good thing to have to go along the next PR for this would be a Suricata-verify PR that used the `cip.class` and `cip.attribute` keywords with values that would not fit a `u8` so we check that this is working as expected.

Reply:
Indeed thanks for the analysis Victor.

We can remove the call to `sscanf` and use `num` instead of `var`

Reply:
So to remove `sscanf` and do `input[i++] = num;` then.

And do an SV test.

Reply:
Yes, and pay attention to use the right integer types uint16 and uint8 ;-)

Reply:
You mean when I'm working on the SV test right?

Reply:
I meant on the Suricata patch

Reply:
Oh! I thought I'm only to modify the `sscanf` part and the `input[i++] = var` part.

Reply:
I think some compilers will warn on `input[i++] = num;`

Reply:
So to change the definition if the `input` array to `uint16_t` then.

Reply:
indeed